R/class-workbook.R

```diff
@@ -5895,24 +5923,13 @@ wbWorkbook <- R6::R6Class(
           visible_sheet_index - 1L
         )
 
-      self$worksheets[[visible_sheet_index]]$sheetViews <-
-        sub(
-          '( tabSelected="0")|( tabSelected="false")',
-          ' tabSelected="1"',
-          self$worksheets[[visible_sheet_index]]$sheetViews,
-          ignore.case = TRUE
-        )
-      if (nSheets > 1) {
-        for (i in setdiff(seq_len(nSheets), visible_sheet_index)) {
-          self$worksheets[[i]]$sheetViews <-
-            sub(
-              ' tabSelected="(1|true|false|0)"',
-              ' tabSelected="0"',
-              self$worksheets[[i]]$sheetViews,
-              ignore.case = TRUE
-            )
-        }
-      }
+      # # TODO what does this code segment do?
+      # self$worksheets[[visible_sheet_index]]$set_sheetview(tabSelected = TRUE)
+      # if (nSheets > 1) {
+      #   for (i in setdiff(seq_len(nSheets), visible_sheet_index)) {
+      #     self$worksheets[[i]]$set_sheetview(tabSelected = FALSE)
+      #   }
+      # }
```

Comment:
Maybe this is actually a required failsafe. Hidden sheets should not be tabSelected.